src/data/action.ts

```diff
@@ -0,0 +1,234 @@
+import {
+  Action,
+  Any,
+  CreateAction,
+  DeleteAction,
+  DiscardAction,
+  EditAction,
+  IdentifiedSanityDocumentStub,
+  PatchOperations,
+  PublishAction,
+  ReplaceDraftAction,
+  UnpublishAction,
+} from '../types'
+import * as validators from '../validators'
+
+/** @internal */
+export class BaseAction {
```

Comment:
This isn't used anywhere? Lots of great API docs here though, could it be integrated with the types instead? Looks like it needs a bit of revising as it's referring to e.g. `commit()` which doesn't seems to stem from an earlier version of this work.